File: Medium/Compare Version Numbers/CompareVersionNumbers.cpp
// https://leetcode.com/problems/compare-version-numbers

class Solution {
public:
  int compareVersion(string version1, string version2) {
    bool isRemaining = true;
    stringstream sv1(version1);
    stringstream sv2(version2);

    while (isRemaining) {
      string sSegmentA, sSegmentB;
      getline(sv1, sSegmentA, '.');
      getline(sv2, sSegmentB, '.');

      int iSegmentA = 0, iSegmentB = 0;

      if (sSegmentA != "") {
        iSegmentA = stoi(sSegmentA);
      }
      if (sSegmentB != "") {
        iSegmentB = stoi(sSegmentB);
      }

      if (iSegmentA != iSegmentB) {
        return iSegmentA < iSegmentB ? -1 : 1;
      }

      isRemaining = sSegmentA != "" || sSegmentB != "";
    }
    return 0;
  }
};

File: Medium/Compare Version Numbers/CompareVersionNumbers.js
// https://leetcode.com/problems/compare-version-numbers

/**
 * @param {string} version1
 * @param {string} version2
 * @return {number}
 */
var compareVersion = function (version1, version2) {
  let v1 = version1.split(".");
  let v2 = version2.split(".");
  let length = v1.length > v2.length ? v1.length : v2.length;
  for (let i = 0; i < length; i++) {
    let segment1 = parseInt(v1.at(i) ?? 0);
    let segment2 = parseInt(v2.at(i) ?? 0);
    if (segment1 != segment2) {
      return segment1 < segment2 ? -1 : 1;
    }
  }
  return 0;
};
